File: sim/sonar/getimu.cpp
#include "msppg.h"

#include <stdio.h>
#include <stdlib.h>

int main(int argc, char ** argv)
{
    if (argc < 3) {
        fprintf(stderr, "Usage:   %s PORT BAUD\n", argv[0]);
        fprintf(stderr, "Example: %s /dev/ttyUSB0 57600\n", argv[0]);
        exit(1);

    }
}

/*
BAUD = 57600

from msppg import MSP_Parser as Parser, serialize_ATTITUDE_Request
import serial

from sys import argv

if len(argv) < 2:

    print('Usage: python3 %s PORT' % argv[0])
    print('Example: python3 %s /dev/ttyUSB0' % argv[0])
    exit(1)

parser = Parser()
request = serialize_ATTITUDE_Request()
port = serial.Serial(argv[1], BAUD)

def handler(pitch, roll, yaw):

    print(pitch, roll, yaw)
    port.write(request)

parser.set_ATTITUDE_Handler(handler)

port.write(request)

while True:

    try:

        parser.parse(port.read(1))

    except KeyboardInterrupt:

        break
*/

File: extras/debug/python/getimu.py
#!/usr/bin/env python3

'''
getimu.py Uses MSPPG to request and handle IMU_SI messages from flight controller

Copyright (C) Simon D. Levy 2019

This file is part of Hackflight.

Hackflight is free software: you can redistribute it and/or modify
it under the terms of the GNU Lesser General Public License as 
published by the Free Software Foundation, either version 3 of the 
License, or (at your option) any later version.
This code is distributed in the hope that it will be useful,     
but WITHOUT ANY WARRANTY without even the implied warranty of
MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
GNU General Public License for more details.

You should have received a copy of the GNU Lesser General Public License 
along with this code.  If not, see <http:#www.gnu.org/licenses/>.
'''

BAUD = 115200

#PORT = 'COM13'          # Windows
PORT = '/dev/ttyACM0' # Linux

from msppg import Parser, serialize_IMU_SI_Request
import serial

class ImuParser(Parser):

    def handle_IMU_SI(self, ax, ay, az, gx, gy, gz):

        print('ax: %+3.3f  ay: %+3.3f  az: %+3.3f | gx: %+3.3f  gy: %+3.3f  gz: %+3.3f ' % (ax, ay, az, gx, gy, gz))
        port.write(request)

if __name__ == '__main__':

    parser = ImuParser()
    request = serialize_IMU_SI_Request()
    port = serial.Serial(PORT, BAUD)

    port.write(request)

    while True:

        try:

            parser.parse(port.read(1))

        except KeyboardInterrupt:

            break

